scio-test/src/test/scala/com/spotify/scio/transforms/AsyncLookupDoFnTest.scala

```diff
@@ -277,8 +268,11 @@ class CallbackFailingScalaLookupDoFn extends ScalaAsyncLookupDoFn[Int, String, A
     )
 }
 
-class TestCacheSupplier extends CacheSupplier[Int, String, java.lang.Long] {
-  override def createCache(): Cache[java.lang.Long, String] =
-    CacheBuilder.newBuilder().build[java.lang.Long, String]()
-  override def getKey(input: Int): java.lang.Long = input.toLong
+// Here we need a custom supplier because guava cache only supports object
+// We can overcome by using a TransformingCache and boxing
+class TestCacheSupplier extends CacheSupplier[Int, String] {
+  override def get(): Cache[Int, String] =
+    new SimpleTransformingCache[Int, java.lang.Integer, String](CacheBuilder.newBuilder().build()) {
+      override protected def transformKey(key: Int): java.lang.Integer = Int.box(key)
+    }
```

Comment:
Here is an example of what can be done if the input type of the `AsyncLookupDoFn` can't be supported by the guava cache.